From e7296be79aa3f2d3eb6c703855eca69015e660e4 Mon Sep 17 00:00:00 2001
From: Piotr Luc <piotr.luc@intel.com>
Date: Mon, 29 Oct 2012 15:08:05 +0100
Subject: Use stride instead width for cleaning YV12 planes in VideorEditor.

To clear entire surface buffer using memset we need take heigh * stride number of bytes
as buffer size for cases where stride is aligned to greater value than width.

Change-Id: Ib4f549ece9836b211e4498d51fb0f447cde78ccc
Signed-off-by: Piotr Luc <piotr.luc@intel.com>
Signed-off-by: Piotr Cwieczkowski <piotr.cwieczkowski@intel.com>
---
 .../lvpp/VideoEditorPreviewController.cpp          |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/libvideoeditor/lvpp/VideoEditorPreviewController.cpp b/libvideoeditor/lvpp/VideoEditorPreviewController.cpp
index 149c4ea..76f1700 100755
--- a/libvideoeditor/lvpp/VideoEditorPreviewController.cpp
+++ b/libvideoeditor/lvpp/VideoEditorPreviewController.cpp
@@ -779,11 +779,11 @@ M4OSA_ERR VideoEditorPreviewController::clearSurface(
      (M4OSA_UInt32)outBufferStride, (M4VIFI_UInt8 *)outBuffer);
 
     /* Fill the surface with black frame */
-    memset((void *)planeOut[0].pac_data,0x00,planeOut[0].u_width *
+    memset((void *)planeOut[0].pac_data,0x00,planeOut[0].u_stride *
                             planeOut[0].u_height * 1.5);
-    memset((void *)planeOut[1].pac_data,128,planeOut[1].u_width *
+    memset((void *)planeOut[1].pac_data,128,planeOut[1].u_stride *
                             planeOut[1].u_height);
-    memset((void *)planeOut[2].pac_data,128,planeOut[2].u_width *
+    memset((void *)planeOut[2].pac_data,128,planeOut[2].u_stride *
                              planeOut[2].u_height);
 
     mTarget->renderYV12();
-- 
1.7.4.1

